 - Kill rtkit_set_realtime() -- it's more trouble than it's worth and 
   it's unnecessary to even have as an option.
 
 - use sched_setscheduler() only for setting RT threads in Wineserver.

 johnstonljordan <AT> gmail <DOT> com

--- a/server/thread.c	2020-07-01 01:34:49.158542000 -0400
+++ b/server/thread.c	2020-07-03 17:48:27.544168903 -0400
@@ -95,58 +95,6 @@ static int dbus_error_to_errno( DBusErro
     return EIO;
 }
 
-static int rtkit_set_realtime( dbus_uint64_t process, dbus_uint64_t thread, dbus_uint32_t priority )
-{
-    DBusConnection* dbus = NULL;
-    DBusMessage *msg = NULL, *rep = NULL;
-    DBusError err;
-    int ret = -1;
-
-    dbus_error_init(&err);
-
-    dbus = dbus_bus_get_private(DBUS_BUS_SYSTEM, &err);
-    if (dbus_error_is_set(&err))
-        goto error;
-
-    dbus_connection_set_exit_on_disconnect(dbus, 0);
-
-    if (!(msg = dbus_message_new_method_call("org.freedesktop.RealtimeKit1",
-                                            "/org/freedesktop/RealtimeKit1",
-                                            "org.freedesktop.RealtimeKit1",
-                                            "MakeThreadRealtimeWithPID")))
-        goto error;
-
-    if (!dbus_message_append_args(msg,
-                                  DBUS_TYPE_UINT64, &process,
-                                  DBUS_TYPE_UINT64, &thread,
-                                  DBUS_TYPE_UINT32, &priority,
-                                  DBUS_TYPE_INVALID))
-        goto error;
-
-    if (!(rep = dbus_connection_send_with_reply_and_block(dbus, msg, -1, &err)))
-        goto error;
-
-    if (dbus_error_is_set(&err))
-        goto error;
-
-    if (dbus_set_error_from_message(&err, rep))
-        goto error;
-
-    ret = 0;
-
-error:
-    if (ret) errno = dbus_error_to_errno(&err);
-    if (rep) dbus_message_unref(rep);
-    if (msg) dbus_message_unref(msg);
-    if (dbus)
-    {
-        dbus_connection_close(dbus);
-        dbus_connection_unref(dbus);
-    }
-    dbus_error_free(&err);
-    return ret;
-}
-
 static int rtkit_set_niceness( dbus_uint64_t process, dbus_uint64_t thread, dbus_int32_t niceness )
 {
     DBusConnection* dbus = NULL;
@@ -963,14 +911,6 @@ int set_thread_priority( struct thread*
         if (sched_setscheduler( thread->fsync_idx, wine_rt_policy|SCHED_RESET_ON_FORK, &param ) == 0)
             return 0;
 #endif
-#ifdef HAVE_DBUS_DBUS_H
-        if (rtkit_set_realtime( thread->unix_pid, thread->unix_tid,
-                                get_unix_priority( thread->process->priority, priority ) ) == 0)
-            return 0;
-        if (rtkit_set_realtime( thread->fsync_idx, thread->fsync_idx,
-                                get_unix_priority( thread->process->priority, priority ) ) == 0)
-            return 0;
-#endif
     }
     else
     {
